doc/GTM_INTEGRATION.md

```diff
@@ -0,0 +1,62 @@
+2. Select the _Templates_ tab in your website's _Google Tag Manager_ workspace.
+3. Click on the _New_ button, select _Import_ in the three-dot menu, and upload the template you downloaded previously.
+4. Review the code and permissions, then save the template.
+
+Then you can add the tag and configure it.
+
+1. Select the _Tags_ tab in your website's _Google Tag Manager_ workspace.
+2. Click on the _New_ button.
+3. Click on the _Triggering_ block and select _Consent Initialization - All Pages_.
+4. Click on the _Tag Configuration_ block and select _Leckerli Cookie Banner_ in the _Custom_ section.
+5. (Optional) Click on _Tag permissions_ and review them.
+6. Set the _leckerli settings_ according to the configuration section below and the [Leckerli README](../README.md). The field must be a valid JSON string and can have multiple lines. You should at least remove not required consent from the permissions array.
+7. (Optional) Change the default consent (this is the consent values before the user's consent settings are updated). The default is set to _denied_ for all consents (_ad_storage_, _analytics_storage_, _functionality_storage_, _personalization_storage_, and _security_storage_).
+8. Click on the _Save_ button and optionally choose another tag name.
+
+### (Optional) Create leckerli trigger event
+
+If you need to fire a tag when the user's consent settings are updated, you can configure a trigger. Remember that tags that do not have required additional consent have already been fired and should not be fired again if you are not sure about what will happen. Tags that have Built-In Consent Checks have probably handled the user's consent update.
+
+1. Select the _Triggers_ tab in your website's _Google Tag Manager_ workspace.
+2. Click on the _New_ button.
+3. Click on the _Trigger Configuration_ block and select _Custom Event_ in the _Other_ section.
+4. Set the _Event name_ value to `leckerliConsentUpdate`.
+5. Click on the _Save_ button and set _Leckerli Consent Update_ as trigger name.
+
+## Configuration
+
+### How to configure leckerli with GTM
+
+* Settings can be chnage in website code and tag settings, website config and tag config are merged (webside have priority)
+* Cookie name is not editable
+* Auto load can be disable in tag config (but not from website), a script will still be loaded to setup callback
+* gtag must be loaded before leckerli
+
+### How to use event trigger
+
+## Update
+
+### How to update the tag template
+re-import and overwrite current tag template, verify if any breaking change between your previous version and the new one exist, fix settings if required. Use preview mode to check that everything works well, then publish.
+
+### a word about versionning
+minor uodate can introduce breaking change for now as leckerli is in active early developement stage, so tag templage will only load updated patch version. The goal is to avoid breaking change in minor update in near future.
```

Comment:
```suggestion
minor update can introduce breaking change for now as leckerli is in active early developement stage, so tag templage will only load updated patch version. The goal is to avoid breaking change in minor update in near future.
```